src/adapter/src/coord/read_policy.rs

```diff
@@ -133,92 +225,65 @@ impl<S: Append + 'static> crate::coord::Coordinator<S> {
-                        .collection(*compute_instance, *id)
-                        .unwrap()
-                        .read_frontier()
-                        .to_owned();
+        // Update the Coordinator's timeline read hold state and organize all id bundles by time.
+        for (timeline, id_bundle) in self.partition_ids_by_timeline(id_bundle) {
+            match timeline {
+                Some(timeline) => {
+                    let TimelineState { oracle, .. } =
+                        self.ensure_timeline_state(timeline.clone()).await;
+                    let read_ts = oracle.read_ts();
+                    let new_read_holds = self.initialize_read_holds(read_ts, id_bundle);
                     let TimelineState { read_holds, .. } =
                         self.ensure_timeline_state(timeline).await;
-                    if !initial_frontier.less_equal(&read_holds.time) {
-                        // This error *SHOULD* be fatal, but it is not currently maintained and cannot merge as an assert.
-                        tracing::error!("Compute collection {:?} (instance {:?}) has read frontier {:?} not less-equal to read_hold.time: {:?}",
-                            id,
-                            compute_instance,
-                            initial_frontier,
-                            read_holds.time,
-                        );
+                    for (time, id_bundle) in &new_read_holds.holds {
+                        id_bundles.insert(Some(*time), id_bundle.clone());
                     }
-                    read_capability
-                        .holds
-                        .update_iter(Some((read_holds.time, 1)));
-                    read_holds
-                        .id_bundle
-                        .compute_ids
-                        .entry(*compute_instance)
-                        .or_default()
-                        .insert(*id);
+                    read_holds.extend(new_read_holds);
+                }
+                None => {
+                    id_bundles.insert(None, id_bundle);
                 }
-
-                self.read_capability.insert(*id, read_capability);
-                compute_policy_updates.push((*id, self.read_capability[id].policy()));
             }
-            self.controller
-                .active_compute()
-                .set_read_policy(*compute_instance, compute_policy_updates)
-                .await
-                .unwrap();
         }
 
-        let mut storage_policy_updates = Vec::new();
-        for id in id_bundle.storage_ids.iter() {
-            let policy = match compaction_window_ms {
-                Some(time) => ReadPolicy::lag_writes_by(time),
-                None => ReadPolicy::ValidFrom(Antichain::from_elem(Timestamp::minimum())),
-            };
+        // Create read capabilities for all objects.
+        for (time, id_bundle) in id_bundles {
+            // We do compute first, and they may result in additional storage policy effects.
```

Comment:
> may result in additional storage policy effects

I don't see this happening. Is this comment outdated now?

Reply:
I think so, I don't really see it happening either. I've removed it.